src/python/espressomd/thermostat.pyx

```diff
@@ -330,39 +343,55 @@ cdef class Thermostat(object):
 
     IF LB_GPU or LB:
         @AssertThermostatType(THERMO_LB)
-        def set_lb(self, kT=None, act_on_virtual=True):
+        def set_lb(
+            self,
+            seed=None,
+            act_on_virtual=True,
+            LB_fluid=None,
+                friction=0.0):
             """
-            Sets the LB thermostat with required parameter 'kT'.
+            Sets the LB thermostat.
 
-            This thermostat requires the feature LB or LB_GPU.
+            This thermostat requires the feature LBFluid or LBFluidGPU.
 
             Parameters
             ----------
-            kT : :obj:`float`
-                 Specifies the thermal energy of the heat bath.
+            LB_fluid : instance of :class:`espressomd.LBFluid` or :class:`espressomd.LBFluidGPU`
+            seed : :obj:`int`
+                 Seed for the random number generator, required
+                 if kT > 0.
             act_on_virtual : :obj:`bool`, optional
                 If true the thermostat will act on virtual sites, default is on.
 
             """
+            valid_LB_fluid = False
+            IF LB:
+                if isinstance(LB_fluid, LBFluid):
+                    valid_LB_fluid = True
+            IF LB_GPU:
+                if isinstance(LB_fluid, LBFluidGPU):
+                    valid_LB_fluid = True
+            if not valid_LB_fluid:
+                raise ValueError(
+                    "The LB thermostat requires a LB / LBGPU instance as a keyword arg.")
 
-            if kT is None:
+            if lb_lbfluid_get_kT() > 0. and not seed:
                 raise ValueError(
-                    "kT has to be given as keyword arg")
-            utils.check_type_or_throw_except(
-                kT, 1, float, "kT must be a number")
-            if float(kT) < 0.:
-                raise ValueError("temperature must be non-negative")
-            global temperature
-            temperature = float(kT)
+                    "seed has to be given as keyword arg")
+
+            if not seed:
```

Comment:
Can you please check that this logic is actually needed. This is somewhat dangerous because setting the temperature to zero resets the rng state. If you do thermalized -> unthermalized -> thermalized, you get the same noise in the second thermalized round, which probably should not happen.

Reply:
Otherwise we would have to check if a seed is needed by checking whether ```Utils::Counter::initial_value() != Utils::Counter::value()```, right?

Reply:
If think it is enough to just leave the seed alone if the temperature is zero. Otherwise you would eg get an error if you set the thermostat twice without integrating.